ui/listner-api/src/main/scala/pl/touk/nussknacker/listner/ListenerManagementFactory.scala

```diff
@@ -0,0 +1,29 @@
+package pl.touk.nussknacker.listner
+
+import com.typesafe.config.Config
+import pl.touk.nussknacker.engine.util.loader.ScalaServiceLoader
+import pl.touk.nussknacker.ui.security.api.LoggedUser
+
+import scala.concurrent.ExecutionContext
+
+trait ListenerManagementFactory {
+  def create(config: Config): ListenerManagement
+}
+
+object ListenerManagementFactory {
+  def serviceLoader(classLoader: ClassLoader): ListenerManagementFactory = {
+    aggregate(ScalaServiceLoader.load[ListenerManagementFactory](classLoader): _*)
+  }
+
+  def aggregate(factories: ListenerManagementFactory*): ListenerManagementFactory = new ListenerManagementFactory {
+    override def create(config: Config): ListenerManagement = {
+      val listeners = factories.map(_.create(config))
+      new ListenerManagement {
+        override def handler(event: ChangeEvent)(implicit ec: ExecutionContext, user: LoggedUser): Unit = {
+          listeners.foreach(_.handler(event))
```

Comment:
What about error handling here or rather on `ListenerManagement` level? Failures in some listener handling shouldn't have influence on other listeners... Also we need to have some logging of failures on top level.

Reply:
now events are handled in a safe way.

Reply:
For aggregates it's ok now, but I think that we need the same in single operation `ListenerManagement` or even upper in `withSideEffect`. Because IMO we don't want to some listener would break our application by throwing Exception. What about changing implementation of `withSideEffect` to:
```
f.onSuccess(f)
f
```
?

Reply:
good point, fixed